src/cmd/bump-versions/main.go

```diff
@@ -38,13 +43,34 @@ func main() {
 		}
 	}
 
+	providers, err := provider.ListProviders(*providerDataDir, logger, ghClient)
+	if err != nil {
+		logger.Error("Failed to list providers", slog.Any("err", err))
+		os.Exit(1)
+	}
+
+	errChan := make(chan error, len(providers))
+
 	for _, p := range providers {
-		slog.Info("Beginning version bump process for provider", slog.String("provider", p.Namespace+"/"+p.ProviderName))
-		err = provider.UpdateMetadataFile(p, *providerDataDir)
+		p := p
+		go func() {
+			errChan <- p.UpdateMetadataFile()
+		}()
```

Comment:
**(todo)** We may want to execute goroutines in batches to avoid running 10k+ goroutines concurrently, especially given that our process will be given 2 vCPU (4 cores/"Machines" for goroutines at most). Overall, concurrency feels a bit preliminary at this point. Did we hit performance bottleneck already? WDYT?

Reply:
Most of them will quickly hit the http rate limiter baked into the github client and be in a low overhead wait state.

Reply:
For a full refresh, it looks like it uses a few % cpu on a single core.